csharp/native/DynamicSubmission/Client/Program.cs

```diff
@@ -0,0 +1,221 @@
+                                                     new CreateResultsRequest.Types.ResultCreate
+                                                     {
+                                                       Data = UnsafeByteOperations.UnsafeWrap(Encoding.ASCII.GetBytes(JsonSerializer.Serialize(input))),
+                                                       Name = "Payload",
+                                                     },
+                                                   },
+                                                 })
+                                  .Results.Single()
+                                  .ResultId;
+
+      // Submit task with payload and result ids
+      var submitTasksResponse = taskClient.SubmitTasks(new SubmitTasksRequest
+                                                       {
+                                                         SessionId = createSessionReply.SessionId,
+                                                         TaskCreations =
+                                                         {
+                                                           new SubmitTasksRequest.Types.TaskCreation
+                                                           {
+                                                             PayloadId = payloadId,
+                                                             ExpectedOutputKeys =
+                                                             {
+                                                               resultId,
+                                                             },
+                                                           },
+                                                         },
+                                                       });
+
+      Console.WriteLine($"Task id {submitTasksResponse.TaskInfos.Single().TaskId}");
+
+      // Wait for task end and result availability
+      await eventClient.WaitForResultsAsync(createSessionReply.SessionId,
+                                            new List<string>
+                                            {
+                                              resultId,
+                                            },
+                                            CancellationToken.None);
+
+      // Download result
+      var result = await resultClient.DownloadResultData(createSessionReply.SessionId,
+                                                         resultId,
+                                                         CancellationToken.None);
+      // Test if the result is equal to the sum of 0 to n 
+      var res            = uint.Parse(Encoding.ASCII.GetString(result));
+      var expectedResult = n * (n + 1) / 2;
+
+      // Throw an exception if the result is wrong
+      if (res != expectedResult)
+      {
+        throw new ArithmeticException($"The result of {n} * ({n} + 1) / 2 is not equal to {Encoding.ASCII.GetString(result)} but to {expectedResult}");
```

Comment:
I think that this message can be made clearer by explicitly saying what is the expected result